S2VX.Game/Story/Note/HoldNote.cs

```diff
@@ -22,20 +23,64 @@ public abstract class HoldNote : S2VXNote {
         [Resolved]
         private S2VXStory Story { get; set; }
 
-        [BackgroundDependencyLoader]
-        private void Load() => AddInternal(SliderPath);
-
-        public Path SliderPath { get; set; } = new() {
+        public SliderPath SliderPath { get; set; } = new() {
             // Show slider path behind note
             Depth = 100,
             Anchor = Anchor.Centre
         };
 
+        [BackgroundDependencyLoader]
+        private void Load() => AddInternal(SliderPath);
+
         protected override void UpdatePosition() {
             UpdateIndicator();
             UpdateSliderPath();
         }
 
+        // These are static helpers for calculating hold related values
+        // TODO: There's probably a better way to share this between HoldApproach and HoldNote for calculating these values
+        // TODO: Optimize these so that they are called once on initialization instead of ran every update loop
+        private static IEnumerable<Vector2> CombineAllCoordinates(Vector2 coordinates, List<Vector2> midCoordinates, Vector2 endCoordinates)
+            => new List<Vector2> { coordinates }.Concat(midCoordinates.Append(endCoordinates));
+        private static float CalculateTotalDistance(IEnumerable<Vector2> allCoordinates) {
+            var totalDistance = 0f;
+            var startCoordinate = allCoordinates.First();
+            foreach (var coordinate in allCoordinates.Skip(1)) {
+                var distance = (coordinate - startCoordinate).LengthFast;
+                totalDistance += distance;
+                startCoordinate = coordinate;
+            }
+            return totalDistance;
+        }
+        public static Vector2 InterpolateCoordinates(
+            double currentTime,
+            double hitTime,
+            double endTime,
+            Vector2 startCoordinates,
+            List<Vector2> midCoordinates,
+            Vector2 endCoordinates
```

Comment:
In-line is my preference, but for this function, adding arguments in-line will cross the 160 character limit.

Reply:
Oh ok, I thought we would just go to a second line then, rather than having one line for each

Reply:
I think it's clearer to me if the arguments are either all in one line or each in a line. Not a fan of putting multiple arguments in a single line since I think it can be easy to missread.